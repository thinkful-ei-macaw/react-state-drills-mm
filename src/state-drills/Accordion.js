import React from "react";

export default class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeSection: null
    }
  }

  static defaultProps = {
    sections: []
  };
  
  handleClick = (title) => {
    this.setState({
      activeSection: title
    })
  }

 
  render() {
    return (
    <ul>
      {this.props.sections.map(item => (
        <li>
        <button onClick={() => this.handleClick(item.title)}>{item.title}</button>
        {this.state.activeSection === item.title ? (<p>{item.content}</p>) : ''}
        </li>
      ))}
      </ul>
    );
  }
}
