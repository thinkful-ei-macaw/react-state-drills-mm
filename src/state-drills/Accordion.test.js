import React from 'react';

import Accordion from './Accordion';

import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe(`Accordion Component`, () => {
  const data = [
    {
      title: "Something",
      content: "This is the content"
    },
    {
      title: "Another thing",
      content: "Some more content pls."
    }
  ];


  it('renders empty li given no sections', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={data} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={data} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('component only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={data} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})