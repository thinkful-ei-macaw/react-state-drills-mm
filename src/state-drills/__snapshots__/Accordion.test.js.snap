// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="Something"
  >
    <button
      onClick={[Function]}
    >
      Something
    </button>
  </li>
  <li
    key="Another thing"
  >
    <button
      onClick={[Function]}
    >
      Another thing
    </button>
    <p>
      Some more content pls.
    </p>
  </li>
</ul>
`;

exports[`Accordion Component component opens a clicked section 1`] = `
<ul>
  <li
    key="Something"
  >
    <button
      onClick={[Function]}
    >
      Something
    </button>
  </li>
  <li
    key="Another thing"
  >
    <button
      onClick={[Function]}
    >
      Another thing
    </button>
    <p>
      Some more content pls.
    </p>
  </li>
</ul>
`;

exports[`Accordion Component renders empty li given no sections 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li
    key="Something"
  >
    <button
      onClick={[Function]}
    >
      Something
    </button>
  </li>
  <li
    key="Another thing"
  >
    <button
      onClick={[Function]}
    >
      Another thing
    </button>
  </li>
</ul>
`;
